### ISSUE TITLE
Update class metrics interface of Precision/Recall/Fbeta to allow calculate them for each individual class

### ISSUE THREAD

**Issue by junwen-austin** [PyTorchLightning/metrics#55]
## 🚀 Feature

I'd like to propose to update class metrics interface of `Precision/Recall/Fbeta` to have the `average` argument include `none` and `weighted` as in the corresponding functional metrics interface.

### Motivation
Current interface with `average` argument restricts to `macro` and `micro`, and because of that one could not use class metrics interface to calculate precision/recall/fbeta for an individual class. For example, in binary classification, one is typically interested in getting metrics results for positive class (class 1) and this cannot be done with the current class interface. Therefore one has go back to the functional metric and this could defeat the purpose of having class metrics (to take care of ddp sync).

On the contrary, sklearn defaults to calculate precision/recall/fbeta for the individual class (class 1) while giving one option to calculate micro/macro/weighted average of these scores.
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.recall_score.html

### Pitch

Update class metrics interface of Precision/Recall/Fbeta to have the `average` argument include `none` and `weighted` as in the corresponding functional metrics interface.

### Alternatives

One can always fall back to the functional metric but I assume this is not what we would like.
 

### Additional context

Really like the new class interface to work with DDP and appreciate all your work!
